modules/data/pending_post.py

```diff
@@ -140,49 +140,68 @@ def get_votes(self, vote: bool) -> int:
-                                      where_args=(self.g_message_id, self.group_id, vote))
+
+        where = "g_message_id = %s and group_id = %s"
+        where_args = (self.g_message_id, self.group_id)
+
+        if vote is not None:
+            where += " and is_upvote = %s"
+            where_args = (self.g_message_id, self.group_id, vote)
+
+        votes = DbManager.select_from(select="admin_id, is_upvote",
+                                        table_name="admin_votes",
+                                        where=where,
+                                        where_args=where_args)
 
         if len(votes) == 0:  # the vote is not present
             return None
 
+        if vote is None:
+            return [(vote['admin_id'], vote['is_upvote']) for vote in votes]
+
         return [vote['admin_id'] for vote in votes]
 
-    def show_admins_votes(self, bot: Bot, approve: bool):
+
+    def show_admins_votes(self, bot: Bot):
         """After a post is been approved or rejected, shows the admins that approved or rejected it \
             and edit the message to delete the reply_markup
 
         Args:
             bot: bot
             approve: whether the vote is approve or reject
         """
-        admins = self.get_list_admin_votes(vote=approve)
-        text = "Approvato da:\n" if approve else "Rifiutato da:\n"
-        tag = '@' if Config.meme_get('tag') else ''
-        for admin in admins or []:
-            username = bot.get_chat(admin).username
-            text += f"{tag}{username}\n" if username else f"{bot.get_chat(admin).first_name}\n"
-
-        remaining_pending_posts = [
-            f"• https://t.me/c/{str(post.group_id)[4:]}/{post.g_message_id}\n"
-            for post in __class__.get_all(group_id=self.group_id)
-            if post.g_message_id != self.g_message_id
-        ]
-        text = f"Spot rimanenti:\n{''.join(remaining_pending_posts)}\n" if remaining_pending_posts else text
-
-        bot.edit_message_reply_markup(chat_id=self.group_id, message_id=self.g_message_id, reply_markup=None)
-        bot.send_message(chat_id=self.group_id, text=text, reply_to_message_id=self.g_message_id)
+
+        # fix circular import
+        get_post_outcome_kb = modules.utils.keyboard_util.get_post_outcome_kb
+
+        bot.edit_message_reply_markup(chat_id=self.group_id,
+                                        message_id=self.g_message_id,
+                                        reply_markup=get_post_outcome_kb(bot, self.get_list_admin_votes()))
+
+        remaining_pending_posts = __class__.get_all(group_id=self.group_id)
+        remaining_pending_posts_count = len(remaining_pending_posts) - 1
+
+        # if there are pending post, reply to the oldest one with the number of remaining pending posts
+        if remaining_pending_posts_count > 0:
+            text = f"⬆️ Post in attesa\nRimangono {remaining_pending_posts_count} post in attesa"
+            oldest_pending_post = remaining_pending_posts[1]
```

Comment:
This is not really a way to get the oldest active post. 
First of all, there is no ordering imposed to the query, although it is pretty safe to assume that the order of insertion should suffice. 
But let me show and example: there are 2 pending posts, A (older) and B (newer). B gets approved. The element `remaining_pending_posts[1]` is still B, so the message sent by the bot will reply to the 'just approved' message B, instead of A (which should be the correct behavior).